### PR TITLE
Prevent migrations from running for database.duplicate

### PR DESCRIPTION
**Pull request by develohpanda**
Once an entity has been loaded from the database, it is already in a valid state - migrations have run, and defaults during `init()` have been set.

Therefore, when duplicating an entity, migrations do **not** need to be run, and the entity does **not** need initialization.

It is seemingly harmless to run `migrate()` and `init()` during duplication, however if the `migrate()` function on the model does complex tasks such as creating other entities, then surplus children can be created.

A duplicate should be just that, a duplicate - it shouldn't have any side effects.

This PR simplifies the duplicate function to remove the `migration()` and `init()` calls from the process, plus adds unit tests to validate these. 😄

### CODE REVIEW

File: packages/insomnia-app/app/common/database.js [hunk 1]
```diff
@@ -483,6 +489,7 @@ export async function docUpdate<T: BaseModel>(
   originalDoc: T,
   ...patches: Array<Object>
 ): Promise<T> {
+  // No need to re-initialize the model during update; originalDoc will be in a valid sate by virtue of loading
```
Comment (gschier) [inline packages/insomnia-app/app/common/database.js]: Typo: "sate" 😝

File: packages/insomnia-app/app/common/database.js [hunk 2]
```diff
@@ -597,13 +604,20 @@ export const duplicate = (database.duplicate = async function<T: BaseModel>(
...
     // 1. Copy the doc
-    const newDoc = Object.assign({}, docToCopy, patch);
-    delete newDoc._id;
-    delete newDoc.created;
-    delete newDoc.modified;
+    const newDoc = Object.assign({}, docToCopy, patch, overrides);
 
-    const createdDoc = await docCreate(newDoc.type, newDoc);
+    // Don't initialize the model during insert, and simply duplicate
+    const createdDoc = await database.insert(newDoc, false, false);
```
Comment (gschier) [inline packages/insomnia-app/app/common/database.js]: Nice! Ended with pretty much the same number of lines 😄

**Review by gschier — DISMISSED** [review]
LGTM if we can get the CI tests working 👍 😄

File: packages/insomnia-app/app/common/__tests__/database.test.js [hunk 3]
```diff
@@ -300,3 +300,58 @@ describe('_fixThings()', () => {
...
+});
+
+describe('docCreate()', () => {
+  beforeEach(globalBeforeEach);
+  afterEach(() => jest.resetAllMocks());
+
+  it('should call migrate when creating', async () => {
+    const spy = jest.spyOn(models.workspace, 'migrate');
+
+    await db.docCreate(models.workspace.type, {
```
Comment (develohpanda) [inline packages/insomnia-app/app/common/__tests__/database.test.js:350]: Fails during `npm run test` but passed when running in IDE. 😢 

image

**Review by gschier — APPROVED** [review]
⭐

**Pull request merged**